src/main/java/com/linkedin/kmf/services/MultiClusterTopicManagementService.java

```diff
@@ -259,11 +254,18 @@ void maybeReassignPartitionAndElectLeader() throws Exception {
           reassignPartitions(zkUtils, brokers, _topic, partitionInfoList.size(), _replicationFactor);
         }
 
+        // Update the properties of the monitor topic if any config is different from the user-specified config
         Properties currentProperties = AdminUtils.fetchEntityConfig(zkUtils, ConfigType.Topic(), _topic);
-        if (!currentProperties.equals(_topicProperties)) {
+        Properties expectedProperties = new Properties();
+        for (Object key: currentProperties.keySet())
+          expectedProperties.put(key, currentProperties.get(key));
+        for (Object key: _topicProperties.keySet())
+          expectedProperties.put(key, _topicProperties.get(key));
+
+        if (!currentProperties.equals(expectedProperties)) {
```

Comment:
No, it won't trigger every time. Note that I am comparing `currentProperties` with `expectedProperties`, where the `expectedProperties` is the union of the user-specified properties (which may be empty) and the `currentProperties`.